src/main/java/org/lantern/multicast/LanternMulticast.java

```diff
@@ -0,0 +1,141 @@
+                addShutdownHook(group, ms);
+            }
+            
+            listen(ms);
+        } catch (final IOException e) {
+            log.error("Error starting multicast?", e);
+        }
+    }
+
+    private void sendHellos(final InetAddress group, final MulticastSocket ms) {
+        final Timer t = new Timer("Multicast-Send", true);
+        t.schedule(new TimerTask() {
+            
+            @Override
+            public void run() {
+                final MulticastMessage mm = 
+                        MulticastMessage.newHello(StaticSettings.getLocalEndpoint());
+                final String msg = JsonUtils.jsonify(mm);
+                final DatagramPacket hi = 
+                    new DatagramPacket(msg.getBytes(Charsets.UTF_8), msg.length(),
+                                        group, sendPort);
+                try {
+                    ms.send(hi);
+                } catch (final IOException e) {
+                    log.warn("Could not send multicast message?", e);
+                }
+            }
+        }, 1000, 10*1000);
+    }
+
+    private void addShutdownHook(final InetAddress group, 
+            final MulticastSocket ms) {
+        Runtime.getRuntime().addShutdownHook(new Thread(new Runnable() {
+
+            @Override
+            public void run() {
+                final Map<String, String> map = new HashMap<String, String>();
+                map.put("type", "bye");
+                map.put("endpoint", StaticSettings.getLocalEndpoint());
+                final String msg = JsonUtils.jsonify(map);
+                final DatagramPacket dp = 
+                        new DatagramPacket(msg.getBytes(Charsets.UTF_8), msg.length(),
+                                            group, sendPort);
+                try {
+                    ms.send(dp);
+                } catch (final IOException e) {
+                    log.error("Could not leave group", e);
+                }
+            }
+            
+        }, "Multicast-Leave"));
+    }
+
+    private void listen(final MulticastSocket ms) {
+        final Runnable run = new Runnable() {
+
+            @Override
+            public void run() {
+                while (true) {
+                    final byte[] buf = new byte[1000];
+                    final DatagramPacket recv = new DatagramPacket(buf, buf.length);
+                    try {
+                        ms.receive(recv);
+                        final MulticastMessage msg = 
+                                JsonUtils.decode(new ByteArrayInputStream(buf), 
+                                        MulticastMessage.class);
+                        if (msg.isBye()) {
+                            endpoints.remove(msg.getEndpoint());
+                            Events.sync(SyncPath.LOCAL_LANTERNS, endpoints);
+                        } else {
+                            endpoints.add(msg.getEndpoint());
+                            Events.sync(SyncPath.LOCAL_LANTERNS, endpoints);
```

Comment:
Could move the 
Events.sync(SyncPath.LOCAL_LANTERNS, endpoints); 
statement outside the conditional.